### DISCUSSION TITLE
Edit queryKeys after fetch data

### DISCUSSION THREAD

**Discussion by tienlx97** [TanStack/query#4167] [Q&A]
Can I edit queryKeys after fetch data. 
i.e: 
- I have a blog Route with path="@:nickName/:slug"   -> `queryKeys=[nickName, slug]`. Suppose slug is unique, so I will fetch blog detail by slug -> data return (blog detail and author nick name) 
- In browser client write: /@tienlx12345/fake-slug-f051e385e87e . Now `queryKeys = ['tienlx12345', 'fake-slug-f051e385e87e']`. But real nickName after fetch is tienlx97 -> I want change queryKeys to `['tienlx97', 'fake-slug-f051e385e87e']`. 

How can I do that ?

**Comment by TkDodo** [comment]
query keys are used as keys for the query cache, if you change it, you would create a new cache entry. You can likely just useQuery with the new query Key and copy over data from the old one with the initialData function.
